src/MongoDB.EntityFrameworkCore/Serializers/SerializationHelper.cs

```diff
@@ -31,13 +31,33 @@ internal static class SerializationHelper
     public static T GetPropertyValue<T>(BsonDocument document, IReadOnlyProperty property)
     {
         var serializationInfo = GetPropertySerializationInfo(property);
-        return ReadElementValue<T>(document, serializationInfo);
+        if(TryReadElementValue(document, serializationInfo, out T value))
+        {
+            return value;
+        }
+
+        if (property.IsNullable)
```

Comment:
Given that TryReadElementValue sets value to default in the false case we could inline this with the other condition,

```
if (TryReadElementValue(document, serializationInfo, out T value) || property.IsNullable) {
  return value;
}

throw...
```